app/jsx/components/AuthorListComp.jsx

```diff
@@ -17,42 +16,60 @@ class AuthorListComp extends React.Component {
   asList = (title, array) => {
     return array.map((x, i) => {
       let c = (i==0) ? "c-authorlist__begin" : (i+1 == array.length) ? "c-authorlist__end" : null
-      if (i==0 && array.length-1>0) {
-        return (<li key={i+x.name} className={c}><span className="c-authorlist__heading">{title}(s):</span> {x.name}&#59; </li>)
-      } else if (i==0 && array.length-1==0) {
-        return (<li key={i+x.name} className={c}><span className="c-authorlist__heading">{title}(s):</span> {x.name} </li>)
+      if (i==0 && array.length-1>0 && title !="Author") {
+        return (<li key={i+x.name} className={c}><span className="c-authorlist__heading">{title}(s):</span> <a href={"/search/?q="+encodeURIComponent("author:"+x.name)}>{x.name}</a>&#59; </li>)
+      } else if (i==0 && array.length-1==0 && title !="Author") {
+        return (<li key={i+x.name} className={c}><span className="c-authorlist__heading">{title}(s):</span> <a href={"/search/?q="+encodeURIComponent("author:"+x.name)}>{x.name}</a> </li>)
       } else if (i<array.length-1) {
-        return (<li key={i+x.name} className={c}>{x.name}&#59; </li> )
+        return (<li key={i+x.name} className={c}><a href={"/search/?q="+encodeURIComponent("author:"+x.name)}>{x.name}</a>&#59; </li> )
       } else {
-        return (<li key={i+x.name} className={c}>{x.name} </li>) 
-      }
+        return (<li key={i+x.name} className={c}><a href={"/search/?q="+encodeURIComponent("author:"+x.name)}>{x.name}</a> </li>) 
+      } 
     })
   }
 
   render() {
     let p = this.props,
-        year = p.pubdate.match(/\d{4}/),
-        source = p.source,
+        year = p.pubdate ? (p.pubdate.match(/\d{4}/)) : null,
+        source = p.source ? p.source : null,
```

Comment:
Isn't 'source = p.source ? p.source : null' same as 'source = p.source'

Reply:
you're right, I was trying to rule out some possibilities, but I'll fix this